README.md

```diff
@@ -33,6 +33,97 @@ Finally, install the build
+      "type": "<old type's name>"
+    }
+  ],
+  "structs": [{
+      "name": "<class/struct name>",
+      "base": "<base class of action if applicable>",
+      "fields": [{
+          "name": "<class field name>",
+          "type": "<class field type>"
+        }
+      ]
+    }
+  ],
+  "actions": [{
+      "name": "<action name>",
+      "type": "<action type>",
+      "ricardian_contract": ""
+    }
+  ],
+  "tables": [{
+   "name": "<table name from multi_index>",
+   "type": "<typename from source file>",
+   "index_type": "i64",
+   "key_names": ["<key name>"],
+   "key_types": ["<key type>"]
+  }],
+  "ricardian_clauses": [{}
+  ],
+  "error_messages": [],
+  "abi_extensions": []
+}
+```
+- Let's unpack what each one these means
+##### abi structure :: types
+- ```types``` is a JSON array of objects, where each object has two fields (`new_type_name` and `type`)
+- If in your smart contract you would like to use a `typedef` of some type to another, simply add those to here. Where `new_type_name` is the new type name and `type` is simply what you are aliasing from.
+##### abi structure :: structs
+- ```structs``` is a JSON array of objects, where each object has three fields ( `name`, `base` and `fields` )
+- If in your smart contract you create a new structure that is not one supplied by `eosiolib`, we are going to express that here.  Some confusion will ultimately arise if using the `EOSIO_ABI` macro and the method style of writing your actions, because they go here too.
+- `name` is the stripped name of the struct/class (i.e. no namespaces, just the name of the class), if using the method style, this is the name of the method.
+- `base` is what class that class inherits from (same thing only the stripped name), or "" if it doesn't inherit from any class.  For the method style this is always "".
+- `fields` is a JSON array of objects with two fields (`name` and `type`), these express the fields of the class we are referencing.  For the method approach, these will be the arguments to your method.
+   - `name` is the field name and `type` is the type of said field.
+##### abi structure :: actions
+- ```actions``` is a JSON array of objects, where each object has three fields (`name`, `type` and `ricardian_contract`)
+- `name` is the name of the action, i.e. the name you would use with `push action`
+- `type` is the type of that action, if you are using the method style and the `EOSIO_ABI` macro, then this is the name of your action method. If you are using the struct approach, then this is simply the name of the struct.
+- `ricardian_contract` is a plain text contract that is cryptographically linked to each action, the full explanation of this is way beyond the scope of this readme.
+##### abi structure :: tables
+- ```tables``` is a JSON array of objects, where each object has five fields (`name`, `type`, `index_type`, `key_names`, `key_types`)
+- `name` is the name of the table that is given the multi_index typedef.
```

Comment:
Could also mention that this is the name that is used in the `table` field of the `get_table_rows` RPC from the chain API plugin (and also the table name used in the `cleos get table` command).